File: backend/src/Ctor.ReportEngine/ReportEngine/src/main/kotlin/com/ctor/reportEngine/db/entities/Company.kt
package com.ctor.reportEngine.db.entities

import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.JoinColumn
import javax.persistence.OneToOne

@Entity(name = "Company")
class Company(@Id var Id: Long, var CompanyName: String, var CompanyLogoId: Long?) {
    @OneToOne
    @JoinColumn(name = "CompanyLogoId", insertable = false, updatable = false)
    var CompanyLogo: CompanyLogo? = null
}

File: backend/src/Ctor.Domain/Entities/Company.cs
using Ctor.Domain.Common;

namespace Ctor.Domain.Entities;
public class Company : BaseEntity
{
    public long CompanyId { get; set; }
    public string CompanyName { get; set; }
    public string Country { get; set; }
    public string City { get; set; }
    public string Address { get; set; }
    public string Email { get; set; }
    public DateTime JoinDate { get; set; }
    public string? Website { get; set; }
    public string? Description { get; set; }
    public long? CompanyLogoId { get; set; }
    public CompanyLogo? CompanyLogo { get; set; }
    public virtual ICollection<Material> Materials { get; set; }
    public virtual ICollection<User> Users { get; set; }
    public virtual ICollection<Vendor> Vendors { get; set; }
    public virtual ICollection<Project> Projects { get; set; }
}
